tests/terminator_tests/improvement_tests/test_evaluator.py

```diff
@@ -60,11 +61,25 @@ def test_evaluate() -> None:
         assert regret_bound == 2.0 * np.sqrt(_get_beta(n_params=1, n_trials=len(trials)))
 
 
+def test_best_value_stagnation_evaluate() -> None:
+    evaluator = BestValueStagnationEvaluator(max_stagnation_trials=1)
+    trials = [create_trial(value=value) for value in [0, 1, 2]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == 1
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
+    trials = [create_trial(value=value) for value in [0, 1, 0]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == 0
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
+    trials = [create_trial(value=value) for value in [0, 0, 0]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == -1
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
```

Comment:
`evaluator.evaluate` always return -1 if the study direction is minimize. Could you verify it returns positive values, please?

Reply:
Added test cases
https://github.com/optuna/optuna/pull/4974/files#diff-09080e058d6651a7a3855dfbf9de39ff4464796da0b2ad025e8e9adac546e7d1R64-R76